### DISCUSSION TITLE
Released v3.2.0-beta.1 with new restore and save actions

### DISCUSSION THREAD

**Discussion by kotewar** [actions/cache#1028] [General]
Hello everyone 👋🏽 

We have released `v3.2.0-beta.1` [version](https://github.com/actions/cache/releases/tag/v3.2.0-beta.1) with implementation of [actions/cache/restore](https://github.com/actions/cache/tree/v3.2.0-beta.1/restore) and [actions/cache/save](https://github.com/actions/cache/tree/v3.2.0-beta.1/save) actions.

It can be used in your workflows as - 
- uses: actions/cache/restore@v3.2.0-beta.1 
- uses: actions/cache/save@v3.2.0-beta.1 

Please do test it out and let us know your feedback. 🙇🏽‍♂️ 

<hr /><em>This discussion was created from the release <a href='https://github.com/actions/cache/releases/tag/v3.2.0-beta.1'>v3.2.0-beta.1</a>.</em>

**Comment by ximon18** [comment]
Works for me. See ([initial run without cache](https://github.com/NLnetLabs/ploutos-testing/actions/runs/3703571823/attempts/2) and then a [subsequent run](https://github.com/NLnetLabs/ploutos-testing/actions/runs/3703571823)). Drilling down to the specific steps where the action is used see [this job](https://github.com/NLnetLabs/ploutos-testing/actions/runs/3703571823/jobs/6275317036) from the initial run and the [this job](https://github.com/NLnetLabs/ploutos-testing/actions/runs/3703571823/jobs/6275818304) from the subsequent run: in particular steps `Fetch cargo-deb from cache`, `Install cargo-deb if needed` and `Force cache save` the latter showing the saving of the cache much earlier than was previously possible, and more importantly saving it before later steps that may fail the job while we still want the results of costly setup steps to be saved to the cache regardless of subsequent failure.

One strange thing is that all of the cache save steps I can see result in `Failed to save` due to the cache key already being in use, but as I manually cleared the caches before the run using `gh actions-cache delete` and that a subsequent run, and even other parallel jobs in the same run, are able to fetch the cache key, I can't see any actions/cache/save or actions/cache post step that actually saved the key in the cache...

**Reply by kotewar** [reply]
Thanks for trying it out @ximon18, I'll take a look at your test runs and get back. :)

**Comment by ximon18** [comment]
@kotewar: For some context, I'm using actions/cache/save to address https://github.com/NLnetLabs/ploutos/issues/47.

**Reply by kotewar** [reply]
Thanks for sharing the context!

**Comment by cdce8p** [comment]
Tested the `cache/restore` action. Seems to work as expected.
Usability wise, I would really love to see an option to fail the workflow if cache restore failed. Using `outputs.cache-hit` works, but it's much more cumbersome than it needs to be. Just an example, in one of my workflow files about 5% (55 out of 950 lines) of all lines are just to exit if the restore step failed.

From the discussion in https://github.com/actions/cache/discussions/1020#discussioncomment-4341251 it's clear, I'm not the only one who would welcome this option. It also doesn't seem to be difficult to implement, as it was part of #1006 initially.

**Comment by henrygab** [comment]
Just writing here to say **_thank you_** for already integrating the feedback from https://github.com/actions/cache/discussions/1020#discussioncomment-4395237 (unconditionally set `cache-matched-key` output on successful restore).

**Reply by kotewar** [reply]
Anytime! :)

Please do try it and let us know if it works as expected.

**Comment by klutzer** [comment]
I've been using the restore action and seems working well 'til now!

Thank you!

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
